wordpress-mobile/WordPress-Android
# Domain dashboard card: Remove feature flag

Fixes #18725 

Removing domain dashboard card, and clean up all related as it is superseded by Plans project.

To test:

- Launch JP app
- `Go` to Me -> Debug settings
- `Verify` no dashboard_card_domain feature flag is present
- `Go` to Home
- `Verify` **Home**, **Menu** options on dashboard all work as before


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:
